packages/evo/src/Evo.ts

```diff
@@ -80,9 +80,68 @@ export class Evo implements Agent {
         },
         LOOP_PREVENTION_PROMPT
       );
+
+      // Allow additional actions to be performed on each value
+      let next = await functionCallLoop.next();
+      while (!next.done) {
+        yield next.value;
+        next = await functionCallLoop.next();
+        await this.condenseFindScriptMessages();
+      }
+      return next.value;
     } catch (err) {
       this.logger.error(err);
       return ResultErr("Unrecoverable error encountered.");
     }
   }
-}
+
+  private async condenseFindScriptMessages(): Promise<void> {
```

Comment:
Hmmmm, I don't like how specific this is. I think we need to find an architectural solution to supporting this feature, rather than specifically going through the chat history for this special case. This path is not very future proof IMO and won't help us in future similar scenarios.

I was thinking that we could add something to the chat message data to help us identify messages that can be "forgotten". Maybe it takes the form of a "tag" or a "lifetime"... will look into this tomorrow.

Reply:
I also don't like it! I was asking for feedback on how to improve this before, and I suggested in the voice chat that we need a better way to identify chat messages. It would be really great to hear your thoughts on how to improve this. I don't always feel comfortable with making too many changes to interfaces/architectures that you guys created and it is really helpful to hear what you think.

Reply:
Makes sense, the path is not immediately clear so need to dig in myself to further provide advice on this. Will get back to you shortly.

Reply:
Left my thoughts here: https://github.com/polywrap/evo.ninja/issues/211#issuecomment-1741876287

I think we should postpone this optimization until benchmarks are all passing.